capabilities/writetarget/write_target.go

```diff
@@ -175,6 +182,48 @@ func success() capabilities.CapabilityResponse {
 	return capabilities.CapabilityResponse{}
 }
 
+// getGasSpendLimit returns the gas spend limit for the given chain ID from the request metadata
+func (c *writeTarget) getGasSpendLimit(request capabilities.CapabilityRequest) (string, error) {
+	spendType := "GAS." + c.chainInfo.ChainID
+
+	for _, limit := range request.Metadata.SpendLimits {
+		if spendType == string(limit.SpendType) {
+			return limit.Limit, nil
+		}
+	}
+	return "", fmt.Errorf("no gas spend limit found for chain %s", c.chainInfo.ChainID)
+}
+
+// checkGasEstimate verifies if the estimated gas fee is within the spend limit and returns the fee
+func (c *writeTarget) checkGasEstimate(ctx context.Context, spendLimit string, report []byte, reportContext []byte, signatures [][]byte, request capabilities.CapabilityRequest) (*big.Int, uint32, error) {
+	// Get gas estimate from ContractWriter
+	fee, err := c.targetStrategy.GetEstimateFee(ctx, report, reportContext, signatures, request)
+	if err != nil {
+		return nil, 0, fmt.Errorf("failed to get gas estimate: %w", err)
+	}
+
+	// Convert spend limit from chain currency to gas units
+	limitFloat, ok := new(big.Float).SetString(spendLimit)
+	if !ok {
+		return nil, 0, fmt.Errorf("invalid gas spend limit format: %s", spendLimit)
+	}
+
+	// Multiply by 10^decimals to convert from chain currency to gas units
+	multiplier := new(big.Float).SetInt(new(big.Int).Exp(big.NewInt(10), big.NewInt(int64(fee.Decimals)), nil))
+	limitFloat.Mul(limitFloat, multiplier)
+
+	// Convert to big.Int for comparison
+	limit := new(big.Int)
+	limitFloat.Int(limit)
+
+	// Compare estimate with limit
+	if fee.Fee.Cmp(limit) > 0 {
+		return nil, 0, fmt.Errorf("estimated gas fee %s exceeds spend limit %s", fee.Fee.String(), limit.String())
```

Comment:
Where is this limit specified? Would this be in the workflow? And if so does the workflow do any sort of check to ensure the user actually has those funds?